lib/Command/FixEncryptedVersion.php

```diff
@@ -0,0 +1,204 @@
+				if ($this->view->is_dir($path)) {
+					$directories[] = $path;
+				} else {
+					$output->writeln("Verifying the content of file $path");
+					$this->verifyFileContent($path, $output);
+				}
+			}
+		}
+	}
+
+	/**
+	 * @param string $path
+	 * @param OutputInterface $output
+	 * @param bool $allowOnce
+	 */
+	private function verifyFileContent($path, OutputInterface $output, $allowOnce = true) {
+		try {
+			if ($this->view->readfilePart($path, 0, 9000) !== false) {
+				$output->writeln("<info>The file $path is: OK</info>");
+			}
+			return true;
+		} catch (HintException $e) {
+			\OC::$server->getLogger()->warning("Issue: " . $e->getMessage());
+			//If allowOnce is set to false, this becomes recursive.
+			if ($allowOnce === true) {
+				//Lets rectify the file by correcting encrypted version
+				return $this->correctEncryptedVersion($path, $output);
+			}
+			return false;
+		}
+	}
+
+	/**
+	 * @param string $path
+	 * @param OutputInterface $output
+	 */
+	private function correctEncryptedVersion($path, OutputInterface $output) {
+		$skipDecrymentedVersion = null;
+
+		$fixedFileInfo = ['filename' => '', 'encryptedVersion' => 0];
+
+		$fileInfo = $this->view->getFileInfo($path);
+		$fileId = $fileInfo->getId();
+		$encryptedVersion = $fileInfo->getEncryptedVersion();
+		$oldEncryptedVersion = $encryptedVersion;
+
+		$storage = $fileInfo->getStorage();
+
+		$cache = new Cache($storage);
+		$fileCache = $cache->get($fileId);
+		$fileCacheStorage = $fileCache->getData()['storage'];
+
+		if (\strpos($fileCacheStorage, "shared::") !== false) {
+			$output->writeln("<info>The file: $path is a share. Hence kindly fix this by running the script under the owner of share</info>");
+			return true;
+		}
+
+		if ($encryptedVersion >= 0) {
+			if ($encryptedVersion === 0) {
+				$skipDecrymentedVersion = true;
+			}
+			//test by decrementing the value till 1 and if nothing works try incrementing
+			if (($skipDecrymentedVersion !== true) && ($fixedFileInfo['filename'] !== $path)) {
+				$encryptedVersion--;
+				while ($encryptedVersion > 0) {
+					if ($fixedFileInfo['filename'] === $path && $fixedFileInfo['encryptedVersion'] > 0) {
+						return true;
+					}
+
+					$cacheInfo = ['encryptedVersion' => $encryptedVersion, 'encrypted' => $encryptedVersion];
+					$cache->put($fileCache->getPath(), $cacheInfo);
+					$output->writeln("<info>Decrement the encrypted version to $encryptedVersion</info>");
+					if ($this->verifyFileContent($path, $output, false) === true) {
+						$output->writeln("<info>Fixed the file $path with version " . $encryptedVersion . "</info>");
+						return true;
+					}
+					$encryptedVersion--;
+				}
+			}
+
+			//So decrementing did not worked. Now lets increment. Max increment is till 25
+			$increment = 1;
+			while ($increment <= 25) {
+				if ($oldEncryptedVersion !== null) {
+					$encryptedVersion = $oldEncryptedVersion + $increment;
+				} else {
+					$encryptedVersion = $encryptedVersion + $increment;
```

Comment:
please recheck, from my understanding this will go as follows:

with $encryptedVersion = 6 at start
 
cycle 1: $increment=1, $encryptedVersion=6+1=7
cycle 2: $increment=2, $encryptedVersion=7+2=9
cycle 3: $increment=3, $encryptedVersion=9+3=12
...

if it doesn't do that (which might be confirmed with unit test), then rewrite the code to make it more clear what happens to the reader

Reply:
Actually it doesn't work like that, I have added php comment. Also adjusted the variable `encryptedVersion` in this section to `$newEncryptedVersion`, the `oldEncryptedVersion` is renamed to `$wrongEncryptedVersion`.
So basically its like this:

when the $wrongEncryptedVersion = 6 at start

cycle 1: $increment=1, $newEncryptedVersion = 6+1 = 7
cycle 2: $increment=1, $newEncryptedVersion = 6+2 = 8
cycle 3: $increment=1, $newEncryptedVersion = 6+3 = 9
Basically the `$wrongEncryptedVersion` remains same and the `$increment` is doing the increment.

Reply:
so it was only a code readability issue, cool

Reply:
Yup.